xonsh/tools.py

```diff
@@ -125,7 +125,14 @@ def subproc_toks(line, mincol=-1, maxcol=None, lexer=None, returnline=False):
             tok.type = 'NEWLINE'
             tok.value = '\n'
             tok.lineno -= 1
-            tok.lexpos = len(line)
+            if len(toks) >= 2:
+                prev_tok_end = toks[-2].lexpos + len(toks[-2].value)
+            else:
+                prev_tok_end = len(line)
+            if '#' in line[prev_tok_end:]:
```

Comment:
Also, is this going to break, for example, on a string that happens to contain `#`?  Or will that make it through alright?